src/watchdog/observers/fsevents.py

```diff
@@ -92,8 +92,29 @@ def on_thread_stop(self):
             self._watch = None
 
     def queue_event(self, event):
-        logger.debug("queue_event %s", event)
-        EventEmitter.queue_event(self, event)
+        # fsevents defaults to be recursive, so if the watch was meant to be non-recursive then we need to drop
+        # all the events here which do not have a src_path / dest_path that matches the watched path
+        if self._watch.is_recursive:
+            logger.debug("queue_event %s", event)
+            EventEmitter.queue_event(self, event)
+        else:
+            if not self._is_recursive_event(event):
+                logger.debug("queue_event %s", event)
+                EventEmitter.queue_event(self, event)
+            else:
+                logger.debug("drop event %s", event)
```

Comment:
Maybe should we prevent logging such line, it will pollute logs, WDYT?

Reply:
I think it's totally fine to keep this on a debug log level as it helps debugging cases when an event is erroneously dropped (or queued). :-)